cmd/lifecycle/exporter.go

```diff
@@ -129,27 +129,70 @@ func (e *exportCmd) Args(nargs int, args []string) error {
 	}
 
 	var err error
-	e.stackMD, e.runImageRef, e.registry, err = resolveStack(e.imageNames[0], e.stackPath, e.runImageRef)
+	e.analyzedMD, err = parseAnalyzedMD(cmd.DefaultLogger, e.analyzedPath)
 	if err != nil {
-		return err
+		return cmd.FailErrCode(err, cmd.CodeInvalidArgs, "parse analyzed metadata")
 	}
 
-	if e.runImageRef == "" {
-		return cmd.FailErrCode(
-			errors.New("-run-image is required when there is no stack metadata available"),
-			cmd.CodeInvalidArgs,
-			"parse arguments",
-		)
+	e.stackMD, err = readStack(e.stackPath)
+	if err != nil {
+		return cmd.FailErrCode(err, cmd.CodeInvalidArgs, "parse stack metadata")
 	}
 
-	e.analyzedMD, err = parseOptionalAnalyzedMD(cmd.DefaultLogger, e.analyzedPath)
+	e.targetRegistry, err = parseRegistry(e.imageNames[0])
 	if err != nil {
-		return cmd.FailErrCode(err, cmd.CodeInvalidArgs, "parse analyzed metadata")
+		return cmd.FailErrCode(err, cmd.CodeInvalidArgs, "parse target registry")
+	}
+
+	// run image
+
+	if !e.supportsRunImage() { // TODO: break out into separate function?
```

Comment:
Would love opinions here

Reply:
I'm a big fan of `naked switch` statements for when clauses become complex like here.

But (_once again, stepping back_), I'm a little more concerned about having to handle these "differences across platform APIs" in unique ways. A separate release branch per API version, let's say, would have it's own problems but is easier grasp at a high-level.

```suggestion
	// ...
	return e.populateRunImageRefIfNeeded()
}

func (e *exportCmd) populateRunImageRefIfNeeded() error {
	failError := func(msg string) error {
		return cmd.FailErrCode(errors.New(msg), cmd.CodeInvalidArgs, "parse arguments")
	}

	switch {
	case e.supportsRunImage() && e.runImageRef != "":
		return nil
	case e.supportsRunImage() && e.runImageRef == "":
		runImageRef, err := e.stackMD.BestRunImageMirror(e.targetRegistry)
		if err != nil {
			return failError("-run-image is required when there is no stack metadata available")
		}

		e.runImageRef = runImageRef
		return nil
	case !e.supportsRunImage() && e.runImageRef == "":
		if e.analyzedMD.RunImage == nil || e.analyzedMD.RunImage.Reference == "" {
			return failError("run image not found in analyzed metadata")
		}

		e.runImageRef = e.analyzedMD.RunImage.Reference
		return nil
	case !e.supportsRunImage() && e.runImageRef != "":
		if e.deprecatedRunImageRef != "" {
			return failError("-image is unsupported")
		}

		return failError("-run-image is unsupported")
	default:
		// this code path shouldn't be reached
		// panic("unhandled case")
	}
}
```

Reply:
> A separate release branch per API version, let's say, would have it's own problems but is easier grasp at a high-level

I believe @jabrown85 is of a similar mindset. Let's continue to discuss as a team!

Reply:
@aemengo thanks for the suggestion! I made some changes in e140e15 - LMK what you think!

Reply:
looks mint! 🌿